Assets/Scripts/Interaction Scripts/ThirdPersonView.cs

```diff
@@ -0,0 +1,450 @@
+                pointCloud.transform.RotateAround(Pivot.transform.position, Vector3.up, angle);
+            }
+        }
+
+
+        private void MoveWorld()
+        {
+            // Negative values are here to make moving around look natural.
+            // Without occlusion this looks mediocre because it seems like the map is being moved in the wrong direction.
+            float moveX = controllerInput.GetLeftThumbDelta().x;
+            float moveZ = controllerInput.GetLeftThumbDelta().y;
+
+            // update map position based on input
+            Vector3 position = World.transform.position;
+
+            if (direction == Direction.REGULAR)
+            {
+                position.x -= moveX * speed * Time.deltaTime * 3.0f;
+                position.z -= moveZ * speed * Time.deltaTime * 3.0f;
+            }
+            else
+            {
+                position.x = moveX * speed * Time.deltaTime * 3.0f;
+                position.z = moveZ * speed * Time.deltaTime * 3.0f;
+            }
+
+
+            World.transform.position = position;
+        }
+
+        private void SetWaypointHeight()
+        {
+            // TODO
+        }
+
+        /// <summary>
+        /// Instantiates and returns a new waypoint at the placePoint position.
+        /// Modifies behavior to add or insert if we are currently colliding with a line
+        /// </summary>
+        /// <param name="groundPoint"> This is the location on the ground that the waypoint will be directly above. </param>
+        /// <returns></returns>
+        private Waypoint PlaceWaypoint(Vector3 groundPoint)
+        {
+            Drone currentlySelectedDrone = WorldProperties.GetSelectedDrone(); // Grabbing the drone that we are creating this waypoint for
+            // Make sure our drone exists
+            if (currentlySelectedDrone != null)
+            {
+                // We will use the placePoint location.
+                Vector3 newLocation = new Vector3(groundPoint.x, groundPoint.y, groundPoint.z);
+                // If we don't have a line selected, we default to placing the new waypoint at the end of the path
+                Waypoint newWaypoint = new Waypoint(currentlySelectedDrone, newLocation);
+                // Add the new waypoint to the drone's path
+                currentlySelectedDrone.AddWaypoint(newWaypoint);
+                // Return the waypoint to announce that we successfully placed one
+                return newWaypoint;
+            }
+            // If we have not added or inserted a waypoint, we need to return null
+            return null;
+        }
+
+
+        private Waypoint MoveWaypoint()
+        {
+            //TODO
+            return null;
+        }
+        
+        /// <summary>
+        /// This method handles the undo and delete functionality
+        /// Removes the waypoint from the scene and from the drone's path
+        /// </summary>
+        public void Undo() /// TODO: add undo for moving around
```

Comment:
is there not a similar function in controllerinteraction.cs line 504 for delete and undo last placement?

Reply:
We are not using ControllerInteractions anymore. Their implementation of the undo functions was somewhat jank (deleted the waypoint by directly modifying the waypoints array).

Reply:
oh we are not? can we delete the controllerInteractions file then?

Reply:
yep, we will tomorrow after the final check.